### ISSUE TITLE
node: Implement checkpoints for recovery procedures

### ISSUE THREAD

**Issue by goshawk-3** [dusk-network/rusk#1094]
#### Summary
In scenarios similar to the one outlined  in https://github.com/dusk-network/rusk/issues/1093, it is essential to have the ability to revert to either the "last epoch" or the most recent "valid" finalized state. 

#### Possible solution design or implementation
Currently, we have a function called `try_revert`, that supports reversion to a single target -  `LastFinalized`. However, this function can be expanded to enable reversion to additional historical checkpoints.
Together with that, we have to either automatically detect&resolve any unrecoverable state or allow the reversion via CLI.

**Comment by autholykos** [comment]
Depends on #1516

**Comment by HDauven** [comment]
Superseded by #1684
